src/main/flight/pid.c

```diff
@@ -52,6 +52,10 @@
 
 extern uint8_t motorCount;
 extern float dT;
+extern bool motorLimitReached;
+bool antiWindupProtection;
+
+#define PREVENT_WINDUP(x,y,z) { if (z) {if (ABS(x) > ABS(y)) { if (x < 0) { x = -ABS(y); }  else { x = ABS(y); } } } else { y = x; }}
```

Comment:
This macro is doing too much things .. it takes more than few minutes to understand.

It should be refactored and heavily commented.

Also there is empty statement (`;`) after this macro call. This may cause problems in some contexts ( `if(test) PREVENT_WINDUP(...); else something();` is syntax error). `do { } while(0)` is usual way to handle this